config/config.exs

```diff
@@ -47,6 +45,7 @@ config :crawly,
   # Stop spider after scraping certain amount of items
   closespider_itemcount: 500,
   # Stop spider if it does crawl fast enough
+  storage_backend: Crawly.DataStorage.FileStorageBackend,
```

Comment:
@Ziinc, I kind of like the idea. As it could simplify the logic of data storage, and everything will be defined as a pipeline. Let me think about it a bit. Maybe I will be able to make a pipeline based alternative and maybe it will be cleaner :)

👍

Reply:
Thinking a bit more about this idea...

I can easily open a file descriptor from the pipeline. But it's not clear how to close it. E.g. the pipeline does not know when the last request was done. And the storage worker does know it. I am still thinking what can be done here, but so far I am out of smart ideas :(

Reply:
Perhaps you could use a "debounce timeout" to close the file, such that each time a new  item comes into the pipeline, it will reset the timeout. Once the timeout is reached, the file can be closed. If new requests come down the pipeline, it would then reopen again.

Reply:
I found an article discussing debouncing with a genserver, might be relevant:

https://bytesize.cultivatehq.com/elixir/otp/nerves/2019/01/03/debounce-with-elixir.html

Reply:
@Ziinc it turns out that if I am killing the process the file descriptor opened by this process is also closed. In general it mean we don't have to call the File.close explicitly. I am currently (during evening time) making the change to switch to the design to Pipelines approach!